Update object tracking examples in docs

ankanpy opened the pull request:
…o detection in a frame.

<!--
Thank you 🙏 for your contribution to [Ultralytics](https://www.ultralytics.com/) 🚀! Your effort in enhancing our repositories is greatly appreciated. To streamline the process and assist us in integrating your Pull Request (PR) effectively, please follow these steps:

1. Check for Existing Contributions: Before submitting, kindly explore existing PRs to ensure your contribution is unique and complementary.
2. Link Related Issues: If your PR addresses an open issue, please link it in your submission. This helps us better understand the context and impact of your contribution.
3. Elaborate Your Changes: Clearly articulate the purpose of your PR. Whether it's a bug fix or a new feature, a detailed description aids in a smoother integration process.
4. Ultralytics Contributor License Agreement (CLA): To uphold the quality and integrity of our project, we require all contributors to sign the CLA. Please confirm your agreement by commenting below:

    I have read the CLA Document and I sign the CLA

For more detailed guidance and best practices on contributing, refer to our ✅ [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Your adherence to these guidelines ensures a faster and more effective review process.
--->


## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary  
Improved YOLO11 tracking documentation and code examples for better clarity and robustness. 🛠️✨  

### 📊 Key Changes  
- Updated YOLO11 tracking code examples to handle cases where no bounding boxes or IDs are detected.  
- Simplified variable naming for better readability (`results` → `result`).  
- Enhanced visualization logic to ensure smoother plotting of tracks and frames.  

### 🎯 Purpose & Impact  
- **Improved Robustness**: Prevents errors when no objects are detected, making the tracking process more reliable.  
- **Better Clarity**: Cleaner and more understandable code examples for users learning to implement YOLO11 tracking.  
- **Enhanced User Experience**: Ensures smoother visualization of tracking results, benefiting developers and end-users alike. 🚀

ankanpy committed:
updated the plotting tracks code sinppet to handle the edge case of no detection in a frame.

UltralyticsAssistant commented:
👋 Hello @ankanpy, thank you for submitting an `ultralytics/ultralytics` 🚀 PR! To ensure a smooth review and integration process, please review the following checklist:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_ — Bruce Lee

For more detailed guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). If your PR addresses a bug, ensure a clear **minimum reproducible example (MRE)** is provided to help evaluate the issue effectively.

An Ultralytics engineer will review your PR soon. Thank you for helping make YOLO11 better and more robust! 🚀✨

ankanpy commented:
Hey @glenn-jocher and team,

I made a small change in the docs readme file:

I have updated the code snippet for the plotting tracks section in the **Track** page to handle the edge case of no detection found in the current frame.

I got this error:
```
Traceback (most recent call last):
  File "/home/ankanxopencv/YOLO/test2.py", line 29, in <module>
    track_ids = results[0].boxes.id.int().cpu().tolist()
                ^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'int'
```
So, I just added a check to make sure the loop will not break :D

Also, I am confused about the CLA where I have to put that text :)

Thanks and regards,
Ankan

RizwanMunawar committed:
Merge branch 'main' into main

RizwanMunawar commented:
Hi @ankanpy! Thanks for the PR, yes, good catch.

Can you please add this sentence as a separate message here?

**I have read the CLA Document and I sign the CLA**

RizwanMunawar committed:
Update track.md

ankanpy commented:
I have read the CLA Document and I sign the CLA

ankanpy commented:
@RizwanMunawar I have added as a seperate massage. Now it's fine I hope!

RizwanMunawar committed:
Update README.md

RizwanMunawar commented:
@ankanpy hi, thanks for signing the CLA.

@glenn-jocher, this PR is good to merge. Thanks

Borda commented on docs/en/modes/track.md:
```diff
@@ -231,14 +231,17 @@ In the following example, we demonstrate how to utilize YOLO11's tracking capabi
+            if results.boxes and results.boxes.id is not None:
+                boxes = results.boxes.xywh.cpu()
+                track_ids = results.boxes.id.int().cpu().tolist()
+            else:
+                pass
```
why empty else?

glenn-jocher replied:
@Borda yes good point, this should be removed @ankanpy

glenn-jocher committed:
Merge branch 'main' into main

RizwanMunawar committed:
remove pass condition

UltralyticsAssistant committed:
Auto-format by https://ultralytics.com

glenn-jocher committed:
Merge branch 'main' into main

glenn-jocher commented:
@RizwanMunawar I think the `results` usage here is not correct, this should be reserved for a list of results like we have in our other examples. Once you take an element this is now a `result` (singular, not plural).

```python
result = results[0]
```

ankanpy commented:
@RizwanMunawar should I help, or you are doing this one!

RizwanMunawar committed:
Update track.md

RizwanMunawar committed:
Update README.md

RizwanMunawar commented:
@ankanpy yes 

> @RizwanMunawar I think the `results` usage here is not correct, this should be reserved for a list of results like we have in our other examples. Once you take an element this is now a `result` (singular, not plural).
> 
> ```python
> result = results[0]
> ```

@glenn-jocher oops, ok got your point. Just updated. I think it's good now!. Thanks

RizwanMunawar committed:
Merge branch 'main' into main

UltralyticsAssistant commented:
🎉 Huge thanks to @ankanpy for the fantastic contribution, with support from @Borda, @RizwanMunawar, and @glenn-jocher! This merge elevates YOLO11 tracking to a whole new level of clarity and robustness. 🚀  

As Henry Ford once said, *"Coming together is a beginning, staying together is progress, and working together is success."* This PR is a testament to the power of collaboration and innovation.  

Your efforts have not only made YOLO11 tracking more reliable but also opened the door for a smoother and more intuitive user experience. We deeply appreciate the passion and dedication each of you brought to this! 🙌
